lxd/instance/drivers/driver_qemu.go

```diff
@@ -1054,28 +1054,31 @@ func (d *qemu) validateRootDiskStatefulStop() error {
 		return err
 	}
 
-	stateDiskSizeStr := pool.Driver().Info().DefaultVMBlockFilesystemSize
-	if rootDiskDevice["size.state"] != "" {
-		stateDiskSizeStr = rootDiskDevice["size.state"]
-	}
+	// Only local root disks need additional checks.
+	if !pool.Driver().Info().Remote {
```

Comment:
This isnt correct. If you're statefully stopping a VM, it doesn't matter if its running on a remote pool or not, it still needs somewhere to save its state file. So these checks should always be done.

Reply:
fixed